Add iOS quick start page (#197)
mongodb/docs-realm: source/ios/quick-start.txt

.. _ios-client-quick-start:

===========
Quick Start
===========

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

> This page contains everything you need to know to get {+client-database+}

Reviewer: ```suggestion
This page contains information to quickly get {+client-database+}
```

integrated into your app. Before you begin, ensure you have:

- :ref:`Created a {+app+} <create-a-realm-app>`
- :ref:`Enabled {+sync+} <sync>`
- :ref:`Installed the iOS SDK <ios-install>`

Import Realm
------------

Near the top of any Swift file that uses Realm, add the following import
statement:

.. code-block:: swift

   import RealmSwift

Initialize the App
------------------

> If you are using {+backend+} features like authentication and sync, you must

Reviewer: ```suggestion
To use {+backend+} features such as authentication and sync, you must
```

access your {+app+} using your {+app+} ID. You can find your {+app+} ID in the

> {+ui+}.

Reviewer: Link here or provide a screenshot

Author: I'm even thinking we could use a very brief but central page for this. We are going to say "Find your App ID" a lot!

.. code-block:: swift

   let app = RealmApp(
       "your-realm-app-id",
       configuration: AppConfiguration(
           baseURL: "https://realm-dev.mongodb.com",
           transport: nil,
           localAppName: nil,
           localAppVersion: nil
       )
   )

Define Your Object Model
------------------------

If have not enabled {+sync+} or you enabled {+sync-short+} with :term:`Developer
Mode` in the {+ui+}, you can define your :ref:`object model <ios-realm-objects>`
directly in code.

.. note::

>    If you have enabled {+sync-short+} but turned off Developer Mode, you can copy and paste the object model definitions that {+backend-short+} generated for you from the :guilabel:`SDKs` tab in the {+ui+}. You must re-enable Developer Mode if you want to make changes to the object model definition from client side code. See :ref:`Configure Your Data Model <configure-your-data-model>`.

Reviewer: I think a screenshot here would be invaluable

.. code-block:: swift

   class Task: Object {

>        @objc dynamic var _id: ObjectId = ObjectId.generate()

Reviewer: Is _id and _partition required for every realm object now, or only those being synced to MongoDB? May want to call this out in a comment.

       @objc dynamic var _partition: String = ""
       @objc dynamic var name: String? = nil
       @objc dynamic var complete = false

       override static func primaryKey() -> String? {
           return "_id"
       }

       convenience init(partition: String, name: String) {
           self.init()
           self._partition = partition
           self.name = name
       }
   }

Authenticate a User
-------------------

> If you have enabled :ref:`anonymous authentication <anonymous-authentication>`, you can log in anonymously to your app:

Reviewer: ```suggestion
Anonymous Authentication
~~~~~~~~~~~~~~~~~~~

Enabling :ref:`anonymous authentication <anonymous-authentication>` allows users to immediately log into your app.
```

Author: Reworked in the vein of suggestion

.. code-block:: swift

   app.login(withCredential: AppCredentials.anonymous()) { (user, error) in
        // Remember to dispatch back to the main thread in completion handlers
        // if you want to do anything on the UI.
        DispatchQueue.main.sync {
          guard error == nil else {
            print("Login failed: \(error!)")
            return
          }

          // Now logged in, do something with user

        }
    }

> If you have enabled :ref:`email/password authentication
> <email-password-authentication>`, you can register a new user with an email and
> password:

Reviewer: ```suggestion
Email/Password
~~~~~~~~~~~

With :ref:`email/password authentication
<email-password-authentication>`, you can register a new user with an email and
password.
```

.. code-block:: swift

   app.usernamePasswordProviderClient().registerEmail(username, password: password, completion: { (error) in
     // Handle errors...
     // Registering just registers. You can now log in.
   }

> If you have enabled :ref:`email/password authentication
> <email-password-authentication>`, you can log in with email and password:

Reviewer: ```suggestion
And you can log in with:
```
